daemon/daemon.go

```diff
@@ -913,12 +881,13 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine, registryService
 	if err != nil {
 		return nil, err
 	}
-	// register graph close on shutdown
-	eng.OnShutdown(func() {
-		if err := graph.Close(); err != nil {
-			logrus.Errorf("Error during container graph.Close(): %v", err)
+	defer func() {
+		if err != nil {
```

Comment:
Only closing graph if there was no error? Don't we want to close no matter what?